packages/@react-aria/utils/src/runAfterTransition.ts

```diff
@@ -36,13 +43,18 @@ function setupGlobalEvents() {
       // The transitioncancel event must be registered on the element itself, rather than as a global
       // event. This enables us to handle when the node is deleted from the document while it is transitioning.
       // In that case, the cancel event would have nowhere to bubble to so we need to handle it directly.
-      e.target.addEventListener('transitioncancel', onTransitionEnd);
+      e.target.addEventListener('transitioncancel', onTransitionEnd, {
+        once: true
+      });
```

Comment:
>but maybe instead of relying on removeEventListener, we should make use of once https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#once

I think applying `once: true` is a good improvement but I think we still need the `removeEventListener` for the cases where the transition is not cancelled.

Reply:
good point, I wish I knew more about when e.target would be null

the comment in the code leads me to believe that even if the element has been removed, we should still get a target. And I can confirm this in https://developer.mozilla.org/en-US/play?id=gyFxtUvVxojQ6wH66%2FqjzMIOXV0TZkRiRXuhuAmB4LOwYajs0BuQYzRf7U5v31B6JKa%2F7is8pzfY%2FIJN just mouse over the box within the 5sec before the box is removed from the dom

if you apply this as a patch in your project, do you still see the problem? do you see any memory leaks like you mentioned possibly existing earlier?

Reply:
I haven't seen evidence of actual memory-leaks but after reading the implementation got bit worried if the event-targets would retained in the `Map` in case of `event.target` being `null` for the transitionEnd phase.

Retaining the reference might prevent browsers garbage collection. See related HTTP203 episode: https://youtu.be/18MRkysjsC0?si=kpL2Db2c1VtexQWZ

However we don't need to address this in the scope of this PR since the original issue for strict-mode compatibility for `event.target` was covered.

Reply:
I'm fine addressing the leak, I agree that it exists. I'm more trying to gauge how bad of a problem it is.

I'm fine doing it as followup, we've not had reports yet of a memory leak. So fingers crossed that means it isn't too bad.

Will continue with the review shortly. I just wanted to ask those questions because in lieu of any tests, your app is the best place to verify the fix works.

Reply:
The funny thing is that our app is not directly relying on this functionality. When something from **react-aria/utils** is imported, this module will run the side-effects of attaching the global-event listeners for **transitionrun** and **transitionend**.

I spent a good amount of time trying to reproduce the case where `event.target` ends up being `null` but wasn't able to find the actual conditions to make it happen. I'm guessing it might be browser vendor + OS specific but I can't confidentially relay on that assumption.

That's why I think the best thing to do here is contain the unit of change just for the strict-mode compliancy which seems to be on your agenda based on the PR-history ✌️.